adapters/mqtt-vertx-base/src/main/java/org/eclipse/hono/adapter/mqtt/PropertyBag.java

```diff
@@ -0,0 +1,99 @@
+/*******************************************************************************
+ * Copyright (c) 2016, 2019 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.hono.adapter.mqtt;
+
+import io.netty.handler.codec.http.QueryStringDecoder;
+import org.eclipse.hono.util.Constants;
+
+import java.time.Duration;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * A collection of methods for processing <em>property-bag</em> set at the end of a topic.
+ *
+ */
+public final class PropertyBag {
```

Comment:
I had expected to find something like
```
static PropertyBag fromTopic(final String topic) {
}
```
which would create a `PropertyBag` instance encapsulating all the properties and providing methods for retrieving property values ...